csgo/addons/sourcemod/scripting/ckSurf/misc.sp

```diff
@@ -3136,3 +3136,36 @@ void resetZone(int zoneIndex)
 	g_mapZones[zoneIndex][Team] = 0;
 	g_mapZones[zoneIndex][zoneGroup] = 0;
 }
+
+void SetPlayerVisible(int client)
+{
+	int iAlpha = GetConVarInt(g_hCvarPlayerOpacity);
+	SetEntityOpacity(client, iAlpha);
+
+	// Render weapons opaque too.
+	int iWeapon = -1, iIndex;
+	while((iWeapon = Client_GetNextWeapon(client, iIndex)) != -1)
+		SetEntityOpacity(iWeapon, iAlpha);
+}
+
+void SetPlayerInvisible(int client)
+{
+	SetEntityRenderMode(client, RENDER_NONE);
+}
+
+void SetEntityOpacity(int ent, int iAlpha)
+{
+	if (iAlpha == 255)
```

Comment:
Well it would be >= 255 if anything. My intention was to use RENDER_NORMAL for 255 and RENDER_NONE for 0 respectively but I commented out the == 0 check because I check for mode != RENDER_NONE in OnSettingsChanged (to avoid setting hidden players visible when noclipping, etc). Otherwise we would need to keep track of player visibility states.

Theoretically RENDER_NORMAL and RENDER_TRANSCOLOR with 255 alpha is virtually the same but I figured it may take more render power? You know I'm a sourcemod/pawn noob coming from a Ruby world of glory and non-semicolons ;-)

I borrowed a little bit from [SMRPG](https://github.com/peace-maker/smrpg/blob/master/scripting/upgrades/smrpg_upgrade_stealth.sp#L297) maybe I should validate iAlpha being between 0-255? And there is also the "prop" part I was talking about which I think is indeed related to hats, etc. I'm not sure why the weapon part is not working but it may be a CS:GO quirk?